### PR TITLE
MODORDERS-817: Associate mapped POLines with single Order based on POLine limit value

### PR DESCRIPTION
**Pull request by Aliaksandr-Fedasiuk**
## Purpose
Each MARC Bib record is mapped in a single POLine. Add a mechanism for associating mapped POLines with a single Order based on the order of the MARC Bib record in the file and the POLine limit value.

## Approach
Thr service is created that will save the orderId if it has not yet been created before or return the orderId if it has already been created.

## Learning
[MODORDERS-817](https://issues.folio.org/browse/MODORDERS-817)

### CODE REVIEW

**Review by RomanChernetskyi — APPROVED** [review]

File: src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java [hunk 1]
```diff
@@ -159,6 +167,24 @@ private Map<String, String> extractOkapiHeaders(DataImportEventPayload dataImpor
     return headers;
   }
 
+  private Future<String> generateSequentialOrderId(DataImportEventPayload dataImportEventPayload, JsonObject tenantConfig) {
+    LOGGER.debug("generateSequentialOrderId :: jobExecutionId: {} ", dataImportEventPayload.getJobExecutionId());
+
+    String tenantId = dataImportEventPayload.getTenant();
+    if (isEmpty(tenantConfig.getString(PO_LINES_LIMIT_PROPERTY))) {
+      return Future.failedFuture(new IllegalArgumentException("Parameter poLines-limit is missing."));
```
Comment (VRohach) [inline src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java:175]: @Aliaksandr-Fedasiuk Let's add in the log at least jobExecutionId

File: src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java [hunk 2]
```diff
@@ -159,6 +167,24 @@ private Map<String, String> extractOkapiHeaders(DataImportEventPayload dataImpor
...
+    }
+    Integer poLinesLimit = Integer.valueOf(tenantConfig.getString(PO_LINES_LIMIT_PROPERTY));
+
+    Record parsedMarcBibRecord = new JsonObject(dataImportEventPayload.getContext().get(MARC_BIBLIOGRAPHIC.value())).mapTo(Record.class);
+    if (parsedMarcBibRecord.getOrder() == null) {
+      return Future.failedFuture(new IllegalArgumentException("Order parameter is missing."));
```
Comment (VRohach) [inline src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java:181]: The same about log.

**Review by VRohach — APPROVED** [review]

File: src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java (hunk 1 above)
Comment (RuslanLavrov) [reply src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java:175]: propose not to fail event processing and instead to use 1 as the default poLines limit, if the parameter is absent (like logic in the mod-orders services)

Comment (Aliaksandr-Fedasiuk) [reply src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java]: done

File: src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java (hunk 2 above)
Comment (Aliaksandr-Fedasiuk) [reply src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java]: done

**Review by SerhiiNosko — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**